crates/tx5/src/proto.rs

```diff
@@ -0,0 +1,580 @@
+//! Types associated with the Tx5 protocol.
+
+use crate::BackBuf;
+use tx5_core::{Error, Result};
+
+/// Tx5 protocol message payload max size.
+pub(crate) const MAX_PAYLOAD: u32 = 0b00011111111111111111111111111111;
+
+/// Protocol version 1 header.
+pub(crate) const PROTO_VER_1: ProtoHeader =
+    ProtoHeader::Version(1, [b't', b'x', b'5']);
+
+/// 4 byte Tx5 protocol header.
+///
+/// The initial 3 bits representing values 0, 1, and 7 are reserved.
+/// Decoders should error on receiving these values.
+#[derive(Debug, PartialEq)]
+pub(crate) enum ProtoHeader {
+    /// This is a protocol version handshake.
+    /// `3 bits == 2`
+    /// `5 bits == version number (currently 1)`
+    /// `24 bytes == as bytes b"tx5"`
+    Version(u8, [u8; 3]),
```

Comment:
I'm confused about how to interpret this. 3 * 8 = 24, but this seems to add up to 3 + 5 + 24 = 32?